### ISSUE TITLE
Extremely large page size on loading https://zeit.co/terms

### ISSUE THREAD

**Issue by ben-ic** [zeit/next.js#1589]
Not a next.js user at the moment, but considering it.
I was wondering why it is that when I load just the terms, which I assume is using next.js, the page size is 4.3MB
[https://tools.pingdom.com/#!/cNiRF4/https://zeit.co/terms](https://tools.pingdom.com/#!/cNiRF4/https://zeit.co/terms)

That seems to me to be extremely large even if it is preloading other pages on the site (especially in the world of mobile data plans).

When I look at the network tab in developer tools, it appears to be preloading JS files for each page (each one is 200KB+).

Thanks,
Ben

**Comment by Ajar-Ajar** [comment]
Is this true?
Looks like a big concern...
Anyone?

**Comment by arunoda** [comment]
Okay. Here's what's happening.

The size of single page is ~250KB and it'll render the page once receives that HTML. 
Then it'll download the `app.js` which is also ~250KB. (This is an async task)
Then it'll prefetch pages in the background in a queue. (It won't affect the normal asset loading in your app)

![screen shot 2017-04-03 at 9 32 13 am](https://cloud.githubusercontent.com/assets/50838/24595153/7692d448-1851-11e7-90a7-ac2da79d7d43.png)

What's cool thing is. When you load another page or the same of the app, (which is very likely) there's no network hits. (other than for the initial HTML)

![screen shot 2017-04-03 at 9 35 40 am](https://cloud.githubusercontent.com/assets/50838/24595172/a9235194-1851-11e7-945c-c0486a5c305d.png)
![screen shot 2017-04-03 at 9 40 09 am](https://cloud.githubusercontent.com/assets/50838/24595176/b754718a-1851-11e7-9ff7-b2e3c88eea97.png)

**Do not judge from the tools like this directly. Do you own research as well.**

**Comment by ben-ic** [comment]
Thanks Arunoda.

That is what I thought is happening as I said in my comment above, I just wanted to confirm.

One thing I want to add is that I don't think we should dismiss new potential users like myself so easily.  When I look at tools I tend to look at how the experts implement them and given that zeit.co are the creators of next.js the first thing I look at is what they are loading and how heavy it is. I shared my concerns when I saw that loading the terms loaded 4.3 MB of additional data which I think is a legit concern as it immediately took me back while investigating next.js for a use case of mine.


Thanks again,
Ben
